hello-data-portal/hello-data-portal-api/src/main/java/ch/bedag/dap/hellodata/portal/initialize/service/DefaultUserInitializer.java

```diff
@@ -71,27 +71,48 @@ public class DefaultUserInitializer {
     @Transactional(propagation = Propagation.REQUIRES_NEW)
     public boolean initDefaultUsers() {
         boolean defaultUsersInitiated = false;
-        if (defaultAdminProperties.getEmail() == null) {
+        String username = defaultAdminProperties.getUsername();
+        String email = defaultAdminProperties.getEmail();
+        if (email == null) {
             log.warn("No default admin properties set, omitting");
             return defaultUsersInitiated;
         }
+        List<UserRepresentation> allUsersFromKeycloak = getAllUsersFromKeycloak();
+        Optional<UserRepresentation> userByEmail = allUsersFromKeycloak.stream().filter(user -> user.getEmail().equalsIgnoreCase(email)).findFirst();
+        Optional<UserRepresentation> userByUsername = allUsersFromKeycloak.stream().filter(user -> user.getUsername().equalsIgnoreCase(username)).findFirst();
+        boolean theSameUser = userByEmail.isPresent() && userByUsername.isPresent() && userByUsername.get().getId().equalsIgnoreCase(userByEmail.get().getId());
+
+        if (!theSameUser) {
+            log.info("Users fetched from the keycloak:");
+            allUsersFromKeycloak.forEach(
+                    userRepresentation -> log.info("Usr {}, username: {}, email: {}", userRepresentation.getId(), userRepresentation.getUsername(), userRepresentation.getEmail()));
+            throw new IllegalStateException(
+                    String.format("There are already two different users in the keycloak for the provided username: %s and the email: %s. Please change the configuration",
+                                  username, email));
+        }
+
         // Check if the default user exists in Keycloak
-        boolean userExistsInKeycloak = getAllUsersFromKeycloak().stream().anyMatch(user -> user.getEmail().equals(defaultAdminProperties.getEmail()));
+        boolean userExistsInKeycloak = allUsersFromKeycloak.stream().anyMatch(user -> user.getEmail().equals(email));
```

Comment:
Couldn't you use here userByEmail.exists() (or if not, adjust to equalsIgnoreCase, like above)?